sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -295,7 +297,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         storage = table.storage.copy(
           locationUri = None,
           properties = storagePropsWithLocation),
-        schema = table.partitionSchema,
+        schema = StructType(EMPTY_DATA_SCHEMA ++ table.partitionSchema),
```

Comment:
Don't we need to add the empty data schema in `newHiveCompatibleMetastoreTable`? In existing `HiveClientImpl`, if two conditions are met `schema.isEmpty` and `HiveExternalCatalog.isDatasourceTable(table)`, we add the empty schema then.

Reply:
according to https://issues.apache.org/jira/browse/SPARK-19279 , I think this new behavior is more reasonable, because empty schema table is problematic to hive and thus not Hive-compatible.  cc @gatorsmile

Reply:
I think this should be good for 2.3, but how about keeping it unchanged in 2.2 for safety? I am just afraid this might break the other library who made an assumption in our metastore.